src/Nethermind/Nethermind.Blockchain/Receipts/IReceiptStorage.cs

```diff
@@ -23,5 +24,6 @@ public interface IReceiptStorage : IReceiptFinder
         void Insert(Block block, params TxReceipt[] txReceipts);
         long? LowestInsertedReceiptBlockNumber { get; set; }
         long MigratedBlockNumber { get; set; }
+        event EventHandler<ReceiptsEventArgs> ReceiptsInserted;
```

Comment:
we create more and ore events, starts to be contagious
it is a different paradigm then our serialized processing

it may be hard to stop @LukaszRozmej

Reply:
I don't really see it as an issue and try to create only when I see a clear  advantage, especially here this allows us to abstract also 'removed' receipts under one easy event.